ce/artifacts/activation.ts

```diff
@@ -26,12 +44,53 @@ export class Activation {
 
     // add tools to the list
     for (const [key, value] of this.tools) {
-      result.push([key, value.fsPath]);
+      result.push([key, value]);
     }
     return result;
   }
 
   get Defines(): Array<[string, string]> {
     return linq.entries(this.defines).toArray();
   }
+
+  get Locations(): Array<[string, string]> {
+    return linq.entries(this.locations).select(([k, v]) => <[string, string]>[k, v.fsPath]).where(([k, v]) => v.length > 0).toArray();
+  }
+
+  get Properties(): Array<[string, Array<string>]> {
+    return linq.entries(this.properties).toArray();
+  }
+
+  /** produces an environment block that can be passed to child processes to leverage dependent artifacts during installtion/activation. */
+  get environmentBlock(): NodeJS.ProcessEnv {
+    const result = this.session.environment;
+
+    // add environment variables
+    for (const [k, v] of this.Variables) {
+      result[k] = v;
+    }
+
+    // update environment paths
+    for (const [variable, values] of this.Paths) {
+      if (values.length) {
+        const s = new Set(values.map(each => each.fsPath));
+        const originalVariable = result[variable] || '';
+        if (originalVariable) {
+          for (const p of originalVariable.split(delimiter)) {
+            if (p) {
+              s.add(p);
+            }
+          }
+        }
+        result[variable] = originalVariable;
+      }
+    }
+
+    // define tool environment variables
+    for (const [key, value] of this.tools) {
+      result[key] = value;
+    }
+
+    return result;
```

Comment:
Wouldn't we also want to include any locations in this environment? Or is this not implemented yet? I would think all aspects of the environment would be included?

Reply:
Well, the locations can be accessed thru `.locations`. I don't want to add them to the CLI environment, because ... I don't know that they'd be used for anything (and just clutter it up)